src/main/java/com/exasol/containers/ExasolContainer.java

```diff
@@ -45,14 +54,16 @@
     private String password = ExasolContainerConstants.DEFAULT_SYS_USER_PASSWORD;
     private final LogPatternDetectorFactory detectorFactory;
     private Set<ExasolService> requiredServices = Set.of(ExasolService.values());
-    private final Set<ExasolService> readyServices = new HashSet<>();
     private final ExaOperation exaOperation;
     private TimeZone timeZone;
-    private boolean isReused = false;
+    private boolean reused = false;
     private final ExasolDockerImageReference dockerImageReference;
     private boolean portAutodetectFailed = false;
     private int connectionWaitTimeoutSeconds = 200;
     private ExasolDriverManager driverManager = null;
+    private final ContainerStatusCache statusCache = new ContainerStatusCache(
```

Comment:
You could use https://www.logicbig.com/how-to/code-snippets/jcode-java-io-files-createtempdirectory.html here